logstash-core/lib/logstash/config/mixin.rb

```diff
@@ -126,6 +146,42 @@ def config_init(params)
     @config = params
   end # def config_init
 
+  # Replace all environment variable references in 'value' param by environment variable value and return updated value
+  # Process following patterns : $VAR, ${VAR}, ${VAR:defaultValue}
+  def replace_env_placeholders(value)
+    if (value.is_a?(String))
+      while value =~ ENV_PLACEHOLDER_REGEX
```

Comment:
I've opened https://github.com/fbaligand/logstash/pull/1 against your fork which implements my suggestion.

It also implements my proposal to fail with an error if the user provides a placeholder that refers to a nonexistent env var and also doesn't provide a default.